TACC/Core-Portal
# task/WP-163 Compress Archive Path Fix

## Overview

The goal for this task was to ensure that `Download Folder` actions in a public system archived the compressed file to the users default private system instead of the public system itself. If no default private system was present then the user will be presented with an error message

## Related

* [WP-163](https://jira.tacc.utexas.edu/browse/WP-163)

## Changes

1- Added a check to determine if the source system is public or private. Based on that, the archive path for the compressed file is set 
2- Added functionality to send back a custom message to `DataFilesDownloadMessageModal` so that the user can be presented with the correct error message 

## Testing

1. Go to Public Data system in core portal 
2. Select a folder and start the download process (make sure the folder is not empty otherwise job will fail)
3. Ensure job starts successfully and that the job output is in the default My Data (Corral) system
4. To test the error message, go to the file `datafiles.sagas.js` and change line 1077 to `const systems = []`
5. Repeat steps 1 & 2 but this time ensure the modal displays the error message: `Folder downloads are unavailable in this portal`

## UI

Job detail with private output path for a public file 
<img width="1910" alt="Screen Shot 2023-08-15 at 1 58 30 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/d4833515-b03e-4585-ab72-e4240df444ee">

Error message 
<img width="1910" alt="Screen Shot 2023-08-14 at 5 25 44 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/50b11655-c5df-4d7d-ae81-4f7acd67cc68">

## Notes

TODO: 
- [x] Add tests for datafiles saga to test the new functionality
